schemas/devContainerTemplate.schema.json

```diff
@@ -0,0 +1,182 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Development Container Template Metadata",
+  "description": "Development Container Template Metadata (devcontainer-template.json). See https://containers.dev/implementors/templates/ for more information.",
+  "definitions": {
+    "Template": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "$schema": {
+          "description": "The JSON schema of the `devcontainer-template.json` file.",
+          "type": "string",
+          "format": "uri"
+        },
+        "description": {
+          "description": "Description of the Template.",
+          "type": "string"
+        },
+        "documentationURL": {
+          "description": "Url that points to the documentation of the Template.",
+          "type": "string",
+          "format": "uri"
+        },
+        "id": {
+          "description": "ID of the Template. The id should be unique in the context of the repository/published package where the Template exists and must match the name of the directory where the devcontainer-template.json resides.",
+          "type": "string"
+        },
+        "keywords": {
+          "description": "List of strings relevant to a user that would search for this Template.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "licenseURL": {
+          "description": "Url that points to the license of the Template.",
+          "type": "string",
+          "format": "uri"
+        },
+        "name": {
+          "description": "Name of the Template.",
+          "type": "string"
+        },
+        "options": {
+          "description": "A map of options that the supporting tools should use to populate different configuration options for the Template.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/Option"
+          }
+        },
+        "platforms": {
+          "description": "Languages and platforms supported by the Template.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "publisher": {
+          "description": "Name of the publisher/maintainer of the Template.",
+          "type": "string"
+        },
+        "version": {
+          "description": "The semantic version of the Template.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "id",
+        "name",
```

Comment:
Hmm. It isn't marked as required in `devContainerFeature.schema.json` now that I look at it, but it is on [the implementors page](https://containers.dev/implementors/features/#devcontainer-feature-json-properties). I'm not yet familiar enough with the spec to know which is accurate, but I suppose the other should be updated to match. 🙂

Reply:
@joshspicer Should we make `name` required in the features schema? Would a feature work if it had no name currently? If not, we would know that making it required in the schema won't break anyone. :)

Reply:
In the tooling `name` is not required.  For example when we build the READMEs in `devcontainers/action` it will use `id` if `name` is not available.

That said, we should probably just make it required since it's useful as a display name in our tooling

One thing to note - if we add `name` as required in the schema for Features, this [validation](https://github.com/devcontainers/action/blob/main/src/utils.ts#L119) code when publishing with the `devcontainers/action` will fail if the Feature is missing `name`.

Reply:
It seems safer to update the spec to make it optional then. We could break existing features as you point out.

One advantage of not making it required is that it is easier to write one-off features for prototyping or testing. We would still want features going into the index to have good names for the UI. 🤷‍♂️